crates/task-impls/src/vid.rs

```diff
@@ -355,6 +364,53 @@ where
                     ))
                     .await;
             }
+
+            HotShotEvent::TransactionsSequenced(payload, _metadata, view) => {
+                // encode transactions for disperse calculation
+                let encoded_txns = match payload.encode() {
+                    Ok(encoded) => encoded,
+                    Err(e) => {
+                        error!("Failed to encode the block payload: {:?}.", e);
+                        return None;
+                    }
+                };
+
+                // TODO <https://github.com/EspressoSystems/HotShot/issues/1686>
+                let srs = test_srs(NUM_STORAGE_NODES);
+                // TODO We are using constant numbers for now, but they will change as the quorum size
+                // changes.
+                // TODO <https://github.com/EspressoSystems/HotShot/issues/1693>
+                let vid = VidScheme::new(NUM_CHUNKS, NUM_STORAGE_NODES, &srs).unwrap();
+                let vid_disperse = vid
+                    .disperse(encoded_txns.into_iter().collect::<Vec<u8>>())
+                    .unwrap();
+
+                // TODO never clone a block
+                // https://github.com/EspressoSystems/HotShot/issues/1858
```

Comment:
I think we can remove this comment?

Reply:
Yeah, looks like we don't clone the block anymore 😃